src/pages/messages/components/NviCandidatePage.tsx

```diff
@@ -50,22 +51,18 @@ export const NviCandidatePage = () => {
     meta: { errorMessage: t('feedback.error.get_registration') },
   });
 
-  const nviListQuery = location.state?.nviQuery;
+  const nviQueryParams = location.state?.nviQueryParams;
   const thisCandidateOffset = location.state?.currentOffset;
 
   const hasOffset = typeof thisCandidateOffset === 'number';
-  const navigateCandidateSearchOffset = hasOffset ? Math.max(thisCandidateOffset - 1, 0) : null;
   const isFirstCandidate = hasOffset && thisCandidateOffset === 0;
 
-  const navigateCandidateQuery = useQuery({
-    enabled: hasOffset,
-    queryKey: ['navigateCandidates', 3, navigateCandidateSearchOffset, nviListQuery],
-    queryFn:
-      navigateCandidateSearchOffset !== null
-        ? () => fetchNviCandidates(3, navigateCandidateSearchOffset, nviListQuery)
-        : undefined,
-    meta: { errorMessage: false },
-    retry: false,
+  if (hasOffset && nviQueryParams) {
+    nviQueryParams.offset = Math.max(thisCandidateOffset - 1, 0);
+  }
+  const navigateCandidateQuery = useFetchNviCandidates({
+    enabled: hasOffset && !!nviQueryParams,
+    params: nviQueryParams ?? {},
```

Comment:
Så lite konfigurasjon som dette tenker jeg er greit da. Spesielt når det har å gjøre med den spesifikke siden som rendres og ikke generelle ting som gjelder for hvert søk uavhengig av url og sidetall.